DragaliaAPI/DragaliaAPI/Features/StorySkip/StorySkipService.cs

```diff
@@ -0,0 +1,191 @@
+    private async Task<QuestMissionStatus> CompleteQuestMissions(int questId, bool[] currentState)
+    {
+        List<AtgenMissionsClearSet> clearSet = new();
+
+        bool[] newState = { true, true, true };
+
+        QuestRewardData rewardData = MasterAsset.QuestRewardData[questId];
+        for (int i = 0; i < 3; i++)
+        {
+            (QuestCompleteType type, int value) = rewardData.Missions[i];
+
+            if (!currentState[i])
+            {
+                newState[i] = true;
+                (EntityTypes entity, int id, int quantity) = rewardData.Entities[i];
+                await rewardService.GrantReward(new Entity(entity, id, quantity));
+                logger.LogDebug("Completed quest mission {missionId}", i);
+                clearSet.Add(new AtgenMissionsClearSet(id, entity, quantity, i + 1));
+            }
+        }
+
+        List<AtgenFirstClearSet> completeSet = new();
+
+        if (currentState.Any(x => !x) && newState.All(x => x))
+        {
+            await rewardService.GrantReward(
+                new Entity(
+                    rewardData.MissionCompleteEntityType,
+                    rewardData.MissionCompleteEntityId,
+                    rewardData.MissionCompleteEntityQuantity
+                )
+            );
+            logger.LogDebug("Granting bonus for completing all missions");
+            completeSet.Add(
+                new AtgenFirstClearSet(
+                    rewardData.MissionCompleteEntityId,
+                    rewardData.MissionCompleteEntityType,
+                    rewardData.MissionCompleteEntityQuantity
+                )
+            );
+        }
+
+        return new QuestMissionStatus(newState, clearSet, completeSet);
+    }
+
+    public async Task IncreaseFortLevels()
+    {
+        Dictionary<FortPlants, FortConfig> fortConfigs = FortConfigurations.FortConfigs;
+
+        foreach (KeyValuePair<FortPlants, FortConfig> keyValuePair in fortConfigs)
+        {
+            FortPlants fortPlant = keyValuePair.Key;
+            FortConfig fortConfig = keyValuePair.Value;
+            await fortRepository.AddToStorage(
+                fortPlant,
+                fortConfig.BuildCount,
+                true,
+                fortConfig.Level
+            );
+        }
+
+        IEnumerable<BuildList> buildListEnum = await fortService.GetBuildList();
+        foreach (BuildList buildList in buildListEnum)
+        {
+            if (fortConfigs.TryGetValue(buildList.PlantId, out FortConfig fortConfig))
+            {
+                DbFortBuild build = await fortRepository.GetBuilding((long)buildList.BuildId);
+                if (build.Level < fortConfig.Level)
+                {
+                    build.Level = fortConfig.Level;
+                    build.BuildStartDate = DateTimeOffset.UnixEpoch;
+                    build.BuildEndDate = DateTimeOffset.UnixEpoch;
+                }
+            }
+        }
+    }
+
+    public async Task IncreasePlayerLevel()
```

Comment:
Does a bit more than increase level, maybe best to rename?
```suggestion
    public async Task UpdateUserData()
```

Reply:
Noted.